.github/workflows/ci-template-check.yml

```diff
@@ -0,0 +1,43 @@
+# Checks if a PR makes any changes that ought to be shared via templating.
+# See SciTools/.github/templates/ for more info.
+
+name: ci-template-check
```

Comment:
Name can be a long string, including spaces, but this is inconsistent with the ones we already have. I am happy to change if preferred. In my demo it was:

```
Check if any changes need sharing via templating
```